src/webgpu/api/validation/resource_usages/textureUsageInRender.spec.ts

```diff
@@ -232,3 +251,113 @@ g.test('shader_stages_and_visibility')
       encoder.finish();
     });
   });
+
+g.test('replaced_binding')
+  .params(poptions('bindingType', kTextureBindingTypes))
+  .fn(async t => {
+    const { bindingType } = t.params;
+    const info = kTextureBindingTypeInfo[bindingType];
+    const bindingTexFormat = info.resource === 'storageTex' ? 'rgba8unorm' : undefined;
+
+    const view0 = t.createTexture().createView();
+    const view1 = t
+      .createTexture({ usage: GPUTextureUsage.STORAGE | GPUTextureUsage.SAMPLED })
+      .createView();
+
+    const bglEntries0: GPUBindGroupLayoutEntry[] = [
+      { binding: 0, visibility: GPUShaderStage.FRAGMENT, type: 'sampled-texture' },
+      {
+        binding: 1,
+        visibility: GPUShaderStage.FRAGMENT,
+        type: bindingType,
+        storageTextureFormat: bindingTexFormat,
+      },
+    ];
+    const bgEntries0: GPUBindGroupEntry[] = [
+      { binding: 0, resource: view0 },
+      { binding: 1, resource: view1 },
+    ];
+    const bindGroup0 = t.device.createBindGroup({
+      entries: bgEntries0,
+      layout: t.device.createBindGroupLayout({ entries: bglEntries0 }),
+    });
+
+    const bindGroup1 = t.createBindGroup(0, view1, 'sampled-texture', undefined);
```

Comment:
For better readability, could we name these as their type? like `sampledView`, `sampledStorageView`, `sampledAndParameterizedEntries`, `sampledAndSampledStorageEntries` etc.. 

This makes it easier to tell what each bind group contains, and what bindings are replaced when we call setBindGroup

Reply:
Good suggestion. I used sampledView and sampledStorageView. For bg/bgl/bgEntries/bglEntries, I still use 0 or 1 to keep it short and consistent among bg/bgl/bgEntries/bglEntries.